### PR TITLE
[READY] Add on-hit effects for lasers and bullet holes for walls from TG

### PR DESCRIPTION
**Pull request by Couls**
**What does this PR do:**
this PR adds on-hit effects for lasers and bullet holes for walls taken from TGstation's sprites
all weapons now also leave bullethole effects on walls in the place they've been shot

**Images of sprite/map changes (IF APPLICABLE):**
ezgif-5-0b451436fd6b

**Changelog:**
:cl:
add: hit effects!
imageadd: hit effect sprites for lasers, bullet hole and dent decals for walls
/:cl:

### CODE REVIEW

**Comment by Ralta** [comment]
Sexy.

**Comment by taukausanake** [comment]
Is there a way to clean the bullet holes off? Some way to patch them back up maybe?

**Comment by Couls** [comment]
> Is there a way to clean the bullet holes off? Some way to patch them back up maybe?

Yes, using a welder on the wall will remove overlays just like repairing them

**Comment by Citinited** [comment]
Do the lasers and bullets create differently-shaped holes?

**Comment by Couls** [comment]
I believe they create the same shaped holes but the effect upon hitting the wall is different

**Comment by Couls** [comment]
If someone wants to sprite seperate shaped holes though I’d be happy to include it

**Comment by variableundefined** [comment]
I've wanted these for so long.

File: code/game/turfs/simulated/walls.dm [hunk 1]
```diff
@@ -434,3 +443,26 @@
 /turf/simulated/wall/narsie_act()
 	if(prob(20))
 		ChangeTurf(/turf/simulated/wall/cult)
+
+/turf/simulated/wall/proc/add_dent(denttype, x=rand(-8, 8), y=rand(-8, 8))
+	if(LAZYLEN(dent_decals) >= 15)
```
Comment (variableundefined) [inline code/game/turfs/simulated/walls.dm]: Put this in a macro like TG does called MAX_DENT_DECALS please. Avoid magic numbers.

Comment (Couls) [reply code/game/turfs/simulated/walls.dm]: fixed

**Pull request merged**